cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -65,17 +65,20 @@
 
   private final Map<Integer, FactTable> resolutionToFactTable;
   private final Map<String, ? extends Aggregation> aggregations;
+  private final Map<String, AggregationAlias> aggregationAliasMap;
```

Comment:
I think `aggregationAliasMap` should not be property of `DefaultCube`. How about having class which extends from `DefaultAggregation`, maintains this mapping, delegates all calls to `DefaultAggregation`, except for `getDimensionNames` where it will return appropriate dimension names after mapping?